### PR TITLE
Improve issue search

### PR DESCRIPTION
**Pull request by ethantkoenig**
Make improvements to issue search: (see https://github.com/go-gitea/gitea/pull/530#pullrequestreview-23520746)
- Add support for searching comment contents
- Add unicode normalization and tokenization
- Move indexer specific code out of models and into a separate module (where it can potentially be reused in the future)
- Update bleve dependency
- Add integration tests for issue search

Don't let the big add/removal figures scare you; that's because of the updated dependency

### CODE REVIEW

**Comment by lunny** [comment]
It seems new issue on sqlite wasn't indexed.

**Comment by ethantkoenig** [comment]
@lunny Fixed

**Comment by lunny** [comment]
LGTM

File: integrations/pgsql.ini.tmpl [hunk 1]
```diff
@@ -10,6 +10,9 @@ PASSWD   = {{TEST_PGSQL_PASSWORD}}
 SSL_MODE = disable
 PATH     = data/gitea.db
 
+[indexer]
+ISSUE_INDEXER_PATH = integrations/indexers/issues.bleve
```
Comment (lafriks) [inline integrations/pgsql.ini.tmpl]: Do not use same path for mysql, pgsql and sqlite as they will conflict when run in parallel

**Review by lafriks — CHANGES_REQUESTED** [review]

**Comment by ethantkoenig** [comment]
@lafriks Good catch, fixed

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
Would be great if these directories would also be added to `make clean` but otherwise LGTM

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
make LG-TM work

**Pull request merged**